Automattic/wp-calypso
# StatsNavigation: do not show the video stats module when the site does not have VideoPress

## Proposed Changes

* Use the `siteHasFeature` selector to determine if VideoPress is present on the current site.
* Use that information to both: disable and turn off the toggle that controls the "videos" module (in the stats navigation section drop-down) and hide the videos module completely if the VideoPress is not present on the current site.
* Still allow users to toggle that module when the site does have VideoPress.

## Why are these changes being made?

The Traffic page is pretty crowded, and the Video module is only relevant when VideoPress is installed, so we should only show it then. Let's default the value to false when the site does not a few the `videopress` feature.

<img width="621" alt="Screenshot 2024-11-28 at 19 58 03" src="https://github.com/user-attachments/assets/62fef0c0-b911-4b60-aea6-6506f9a3499c">
<img width="598" alt="Screenshot 2024-11-28 at 19 58 26" src="https://github.com/user-attachments/assets/e657bfc1-6086-4eab-8ad9-1c4c45a1dc50">


## Testing Instructions

1. Pick two sites - one that have the VideoPress plan purchased (i.e. has the `videopress` features) and one that does not have it. I'm going to use two of my personal sites: wptest.macbre.net (site A, with the VideoPress enabled) farerskiekadry.pl (site B, without VideoPress).
2. Verify that these test sites have the expected features list by calling WP.com API endpoint:

<img width="525" alt="Screenshot 2024-11-28 at 14 26 10" src="https://github.com/user-attachments/assets/ebc766c8-5c46-41e6-a5ad-d18656f1a849">

3. Now, visit the stats section in your local Calypso instance (with this fix checked out):

* http://calypso.localhost:3000/stats/day/farerskiekadry.pl
* http://calypso.localhost:3000/stats/day/wptest.macbre.net

4. You can check the `StatsNavigation` component properties in the React dev tools:

<img width="557" alt="Screenshot 2024-11-28 at 20 54 07" src="https://github.com/user-attachments/assets/5e3335a1-daef-4930-a753-61ec6e218104">

> Note the `isVideoPress` property is set to `false` when viewing stats for the site without VideoPress.

And the following now renders as module switches:

<img width="557" alt="Screenshot 2024-11-28 at 23 18 36" src="https://github.com/user-attachments/assets/49471037-f95d-46fc-ad21-8cd297426168">

And when the video module is available for the site:

<img width="557" alt="Screenshot 2024-11-28 at 23 31 22" src="https://github.com/user-attachments/assets/c19b09a1-4c19-48f7-bef8-9e1b04779346">


## Pre-merge Checklist
